### PR TITLE
Move set of EKF origin down into GCS_MAVLINK

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1854,6 +1854,34 @@ MAV_RESULT GCS_MAVLINK::handle_command_camera(const mavlink_command_long_t &pack
...
+    // check location is valid
+    if (!check_latlng(loc)) {
+        return;
+    }
+
+    AP_AHRS &ahrs = AP::ahrs();
+
+    // check if EKF origin has already been set
+    Location ekf_origin;
+    if (ahrs.get_origin(ekf_origin)) {
```
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Should we really do this? I feel like this should be up to AHRS/EKF to decide.

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -883,7 +883,7 @@ void GCS_MAVLINK_Plane::handleMessage(mavlink_message_t* msg)
                 }
                 plane.ahrs.set_home(new_home_loc);
                 AP::ahrs().set_home_status(HOME_SET_NOT_LOCKED);
-                plane.Log_Write_Home_And_Origin();
+                AP::ahrs().Log_Write_Home_And_Origin();
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply ArduPlane/GCS_Mavlink.cpp]: I'm being consistent with the lines around it.

I have more commits here which *do* canonicalise these - by moving them all elsewhere.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: I don't know.  If we do change it then 6 months later we'll probably add a force option.

Comment (OXINARF) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: You mean... if we **keep it** then 6 months later we'll probably add a force option.

Don't forget this is quite new code so this isn't exactly assumed behavior.

File: ArduPlane/GCS_Mavlink.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (OXINARF) [reply ArduPlane/GCS_Mavlink.cpp]: Lines around it aren't consistent... There is a `plane.ahrs` followed by a `AP::AHRS()` right above it.

**Comment by tridge** [comment]
looks good, thanks

**Pull request merged**